wallet/chain/p/builder/builder_test.go

```diff
@@ -0,0 +1,183 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+//nolint:gosec // This file does not need cryptographically strong randomness
+package builder
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils"
+	"github.com/ava-labs/avalanchego/vms/components/avax"
+	"github.com/ava-labs/avalanchego/vms/components/verify"
+	"github.com/ava-labs/avalanchego/vms/platformvm/stakeable"
+	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
+)
+
+func generateUTXOs(assetID ids.ID, locktime uint64) []*avax.UTXO {
+	utxos := make([]*avax.UTXO, rand.Intn(10))
+	for i := range utxos {
+		var output avax.TransferableOut = &secp256k1fx.TransferOutput{
+			Amt: rand.Uint64(),
+			OutputOwners: secp256k1fx.OutputOwners{
+				Locktime:  rand.Uint64(),
+				Threshold: 1,
+				Addrs:     []ids.ShortID{ids.GenerateTestShortID()},
+			},
+		}
+		if locktime != 0 {
+			output = &stakeable.LockOut{
+				Locktime:        locktime,
+				TransferableOut: output,
+			}
+		}
+		utxos[i] = &avax.UTXO{
+			UTXOID: avax.UTXOID{
+				TxID:        ids.GenerateTestID(),
+				OutputIndex: rand.Uint32(),
+			},
+			Asset: avax.Asset{
+				ID: assetID,
+			},
+			Out: output,
+		}
+	}
+	return utxos
+}
+
+func TestSplitUTXOsByLocktime(t *testing.T) {
+	var (
+		require = require.New(t)
+
+		unlockedTime               uint64 = 100
+		expectedUnlockedUTXOsPart0        = generateUTXOs(ids.GenerateTestID(), 0)
+		expectedUnlockedUTXOsPart1        = generateUTXOs(ids.GenerateTestID(), unlockedTime)
+		expectedLockedUTXOsPart0          = generateUTXOs(ids.GenerateTestID(), unlockedTime+100)
+		expectedUnlockedUTXOsPart2        = generateUTXOs(ids.GenerateTestID(), unlockedTime-1)
+		expectedLockedUTXOsPart1          = generateUTXOs(ids.GenerateTestID(), unlockedTime+1)
+
+		utxos = utils.Join(
+			expectedUnlockedUTXOsPart0,
+			expectedUnlockedUTXOsPart1,
+			expectedLockedUTXOsPart0,
+			expectedUnlockedUTXOsPart2,
+			expectedLockedUTXOsPart1,
+		)
+		expectedUnlockedUTXOs = utils.Join(
+			expectedUnlockedUTXOsPart0,
+			expectedUnlockedUTXOsPart1,
+			expectedUnlockedUTXOsPart2,
+		)
+		expectedLockedUTXOs = utils.Join(
+			expectedLockedUTXOsPart0,
+			expectedLockedUTXOsPart1,
+		)
+	)
+
+	utxosWithAssetID, utxosWithOtherAssetID := splitUTXOsByLocktime(utxos, unlockedTime)
+	require.Equal(expectedUnlockedUTXOs, utxosWithAssetID)
```

Comment:
This is an example of where I strongly believe that `assert` should be used over `require`. The failure of this comparison doesn't necessarily corrupt the next one, and it may very well be useful to know if/how they fail simultaneously.

Same applies to the next test.